ui/src/main/resources/edu/wpi/grip/ui/templates/python/macros.vm

```diff
@@ -0,0 +1,34 @@
+#macro( input $inp $c)
+self.__$inp.name()$c#end
+
+
+#macro( newInput $inp $c)
+#if($inp.type().contains("Type"))
+#input($inp $c) = ${inp.type()}.$inp.value().replaceAll(' ','_')#elseif($inp.hasValue())
+#if($inp.value().contains("source"))
+#input($inp $c) = self.__$inp.value()#else
+#input($inp $c) = $inp.value()#end
+#else##connected socket
+#input($inp $c) = self.__$inp.value()#end#end
+
+#macro( needs $enum )
+#set($BooleanThatDoesNotMatter = $enums.add($enum))#end
```

Comment:
It is language specific. The classes are mostly non language specific. The exception being the template methods which inherit from an abstract class and adding a method because one language needs it does not make sense.